File: 프로그래머스/lv0/120897. 약수 구하기/약수 구하기.java
import java.util.ArrayList;
import java.util.Comparator;

class Solution {
    public ArrayList solution(int n) {
        ArrayList<Integer> list = new ArrayList<>();
        for(int i=1; i<=n; i++) {
            if(n%i == 0) {
                list.add(i);
            }
        }
        return list;
    }
}
/*
    접근법 :
    가변배열을 설정해 for문을 통해 나온 약수들을 넣어줌

    다른 접근법 :
    1. 주어진 수의 절반을 대상으로 확인
        약수는 본인을 제외하고 n/2 보다 클 수 없기 때문에... -> 성능이 좋아진다.
        
    class Solution {
        public ArrayList solution(int n) {
        ArrayList<Integer> list = new ArrayList<>();
            for(int i=1; i<=n/2; i++) {
                if(n%i == 0) {
                    list.add(i);
                }
            }
        list.add(n);
        return list;
        }
    }
*/

File: 프로그래머스/lv0/120897. 약수 구하기/약수 구하기.js
function solution(n) {
    let result = [];
    for(let i=1; i<=n; i++) {
        if(n%i === 0) {
            result.push(i);
        }
    }
    return result.sort( (a, b) => {return a-b});
}
/*
    접근법 :
    : n의 약수를 구한다. -> for문의 조건식을 통해 약수를 구한다.
    : 배열에 담는다. -> if문을 이용해 약수일 때 배열에 넣어줌
    : 오름차순으로 반환 한다. -> sort();를 이용해 오름차순 반환


    다른 접근법 :
    1. 주어진 수의 절반을 대상으로 확인
        약수는 본인을 제외하고 n/2 보다 클 수 없기 때문에... -> 성능이 좋아진다.

    function solution(n){
        let result = [];
        for(let i=1; i<=n/2; i++) {
            if(n%i === 0) {
                result.push(i);
            }
        }
        return [...result, n];
    }
*/